src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +17,176 @@ import {
 
 import { User } from "types/User";
 
-// @debt remove styled-components in favour of using our standard scss patterns
-import * as S from "./UserProfilePicture.styles";
 import "./UserProfilePicture.scss";
 
+const randomAvatarUrl = (id: string) =>
+  "/avatars/" +
+  RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
+
+// TODO: refactor this to accept named props instead of positional args
+const avatarUrl = (
+  id: string,
+  anonMode?: boolean,
+  pictureUrl?: string,
+  miniAvatars?: boolean
+): string => {
+  if (anonMode || !id) {
+    return DEFAULT_PROFILE_IMAGE;
+  }
+
+  if (!miniAvatars && pictureUrl) {
+    return pictureUrl;
+  }
+
+  // TODO: how is this intended to be used? Why doesn't it use the profile image? It seems to be something set on venue config..
+  if (miniAvatars) {
+    return randomAvatarUrl(id);
+  }
```

Comment:
[TODO]: how is `miniAvatars` intended to be used? Why doesn't it use the profile image? It seems to be something set on venue config..

Reply:
Potentially OOS to change this